source/DSCResources/DSC_CMAssetIntelligencePoint/DSC_CMAssetIntelligencePoint.psm1

```diff
@@ -0,0 +1,481 @@
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Yes')]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter()]
+        [String]
+        $SiteServerName,
+
+        [Parameter()]
+        [String]
+        $CertificateFile,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $Schedule,
+
+        [Parameter()]
+        [Boolean]
+        $Enable,
+
+        [Parameter()]
+        [Boolean]
+        $EnableSynchronization,
+
+        [Parameter()]
+        [Boolean]
+        $RemoveCertificate,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -IsSingleInstance $IsSingleInstance
+
+    try
+    {
+        if (($Schedule) -and ($EnableSynchronization -eq $false))
+        {
+            throw $script:localizedData.ScheduleNoSync
+        }
+
+        if (($CertificateFile) -and ($RemoveCertificate -eq $true))
+        {
+            throw $script:localizedData.CertMismatch
+        }
+
+        if ($Ensure -eq 'Present')
+        {
+            if ($state.Ensure -eq 'Absent')
+            {
+                if (-not $PsBoundParameters.ContainsKey('SiteServerName'))
+                {
+                    throw $script:localizedData.ServerNameAdd
+                }
+
+                if ($null -eq (Get-CMSiteSystemServer -SiteCode $SiteCode -SiteSystemServerName $SiteServerName))
+                {
+                    Write-Verbose -Message ($script:localizedData.SiteServerRole -f $SiteServerName)
+                    New-CMSiteSystemServer -SiteCode $SiteCode -SiteSystemServerName $SiteServerName
+                }
+
+                Write-Verbose -Message ($script:localizedData.AddAPRole -f $SiteServerName)
+                Add-CMAssetIntelligenceSynchronizationPoint -SiteSystemServerName $SiteServerName
+            }
+
+            $evalList = @('CertificateFile','Enable','EnableSynchronization')
+
+            foreach ($param in $PSBoundParameters.GetEnumerator())
+            {
+                if ($evalList -contains $param.key)
+                {
+                    if ($param.Value -ne $state[$param.key])
+                    {
+                        Write-Verbose -Message ($script:localizedData.SettingValue -f $param.Key, $param.Value)
+                        $buildingParams += @{
+                            $param.Key = $param.Value
+                        }
+                    }
+                }
+            }
+
+            if (($RemoveCertificate) -and (-not [string]::IsNullOrEmpty(($state.CertificateFile))))
```

Comment:
extra ( ) on $state.CertificateFile
if (($RemoveCertificate) -and (-not [string]::IsNullOrEmpty($state.CertificateFile)))

Reply:
removed